### ISSUE TITLE
How to add a new backend / runtime on Elyra for a new batch scheduler ?

### ISSUE THREAD

**Issue by andrewssobral** [elyra-ai/elyra#1691]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
I would like to ask how to add a new runtime for Elyra Pipeline execution (e.g. SLURM, Proactive, PBS, etc...)

**Describe alternatives you've considered**
This is the Python SDK to run a pipeline on the Proactive Job Scheduler
https://github.com/ow2-proactive/proactive-python-client
How could I integrate it to be able to run Elyra Pipelines on Proactive? 
There's a kind of tutorial / documentation for that?
There's a kind of code template I can follow?
I would like to have a kind of 
```
$ elyra-pipeline submit demo.pipeline  --runtime-config proactive
```

**Additional context**
Proactive Job Scheduler: https://www.activeeon.com/products/workflows-scheduling/

**Comment by kevin-bates** [comment]
Hi @andrewssobral - thank you for raising this issue.  We've started asking ourselves this same question: https://github.com/elyra-ai/elyra/discussions/1672. :smile:

Although the linked discussion lists two areas that a third-party runtime provider would need to satisfy, there is a third that I've realized while reviewing pull request https://github.com/elyra-ai/elyra/pull/1620#discussion_r633845759.  These three areas can be summarized as follows:

1. Runtime-specific schema that defines the necessary metadata types for interacting with the runtime.
2. Runtime processor implementation that uses the values of the runtime schema to interact with the runtime relative to job submission and export capabilities.
3. Component parsers that _normalize_ runtime-specific components into JSON that the frontends can consume in a (preferably) runtime-independent manner. 

Although a general POC could be accomplished using the first two items today via manual processes (and analyzing existing code), we really need to come up with a well-defined mechanism for how "Runtime Providers" will go about implementing and registering their packages for discovery and integration.  As a result, we invite you to the linked discussion item as well as our [Dev Community Call - which takes place every Thursday at 9am PDT](https://github.com/elyra-ai/elyra#weekly-dev-community-meeting) - where we'd be happy to get your ideas and feedback.
